example.cpp

```diff
@@ -363,6 +364,59 @@ void logicsig() {
   std::cout << "logicsig pass" << std::endl;
 }
 
+void multisig() {
+
+  //These mnemonics generate the msig_address below.
+  //Only use this on the testnet
+  auto mnemonic1 = R"(base giraffe believe make tone transfer wrap attend
+                      typical dirt grocery distance outside horn also abstract
+                      slim ecology island alter daring equal boil absent
+                      carpet)";
+
+  auto mnemonic2 = R"(use animal lonely tragic style wealth uniform poverty 
+                      idle sail rice dutch patient sport start shine
+                      stem path client plunge mutual achieve border absent 
+                      aspect)";
+
+
+  const std::string msig_address = "MZVMI5WFABIGZKZEBEWIWEEFQCUE4HVOBUTTYFOYGXGZVZAJNFD6PVQHX4";
+
+  std::vector<Address> addresses; 
+  std::vector<Account> accounts;
+  accounts.push_back(Account::from_mnemonic(mnemonic1));
+  accounts.push_back(Account::from_mnemonic(mnemonic2));
+  addresses.push_back(accounts[0].public_key());
+  addresses.push_back(accounts[1].public_key());
+  MultiSig msig{addresses};
+
+  //Verify Multisig Public Address is expected Address
+  assert(msig_address == msig.address());
+
+  AlgodClient client{};
+  auto resp = client.params();
+  assert(resp.status == 200); //Verify we communicate with node before proceeding
+  const auto& suggested = *resp.json;
+
+  //Verify Multisig Txn can be signed and sent to the "to" address
+  Address to{"CX5FN4QNPFBVILSTMBA7URZXQQPP3IZWAQ5OAENHQKIPIYP4CESAQ77PJA"};
+  Transaction t = Transaction::payment(msig.address(),
+                                       to, 12345, {},
+                                       suggested["min-fee"].GetUint64(),
+                                       suggested["last-round"].GetUint64()+1,
+                                       suggested["last-round"].GetUint64()+1001,
+                                       suggested["genesis-id"].GetString(),
+                                       b64_decode(suggested["genesis-hash"].GetString()),
+                                       {}, {}, {});
+
+  //Create signed Multisig Txn using a 
+  //collection of secret_keys
+  auto smsig = t.sign(msig.sign(accounts)); 
+  resp = client.submit(smsig);
+  assert(resp.status == 200);
```

Comment:
This is a great test.  Can you also add tests in the style of the other signatures that don't need a running algod?  They would create the transaction and compare it to a "golden" decoding that was produced by a known good SDK.  I grabbed the ones I used from the python sdk, but I believe all have them.

Reply:
Great suggestion. I'll go ahead and add this. Should I remove the online test then so that there are no dependencies on an active node in order to checkout/regression test the SDK?

Reply:
I think it would be reasonable to have a command-line accessible way to use multisig stuff that hits algod, but I also think that would be a lot of argument processing. Have a look at what I added for looking at accounts and such, and see if you want to add a similar function.  No needed for this PR though.